### PR TITLE
Scannergates

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
XGWvtJbPH9

Scanner Gates can now have assemblies attached to them! Allow you to trigger anything you want with them!

There are 6 wires on them now.
Successful Scan
Failed Scan
Any Scan
Toggle Signaling
Disable Signaling
Enabling Signaling

<will attach video>

## Why It's Good For The Game
Imagine the possibilities! Traps! Security Checkpoints! 
 

## Changelog
:cl:
add: Scanner Gates can now trigger assemblies!
/:cl:

Please Close PR #52508

### CODE REVIEW

**Comment by Axanthic** [comment]
Added a general trigger wire and renamed some of the light variables.

**Comment by Axanthic** [comment]
Condensed some of the code a bit better. Removed the icon state check - Will re-add if anyone wants to make a sprite for it.

File: code/game/machinery/scan_gate.dm [hunk 1]
```diff
@@ -27,6 +27,10 @@
 	idle_power_usage = 50
 	circuit = /obj/item/circuitboard/machine/scanner_gate
 
+	var/pass = FALSE
+	var/fail = FALSE
+	var/triggered = FALSE
+	var/ignore_signals = FALSE
```
Comment (Rohesie) [inline code/game/machinery/scan_gate.dm:33]: Could you do a `scanner_flags` variable and turn these booleans into bitfields?

File: code/datums/wires/scan_gate.dm [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/datum/wires/scan_gate
+	holder_type = /obj/machinery/scanner_gate
+	proper_name = "Scanner Gate"
+
+/datum/wires/scan_gate/New(atom/holder)
+	wires = list(
+		WIRE_PASS, WIRE_FAIL, WIRE_DISABLE, WIRE_ACTIVATE, WIRE_DISARM, WIRE_PROCEED
```
Comment (Rohesie) [inline code/datums/wires/scan_gate.dm:7]: Why is this on `New()` instead of in the datum definition?

File: code/datums/wires/scan_gate.dm [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+			V.ignore_signals = FALSE
+
+/datum/wires/scan_gate/get_status()
+	var/obj/machinery/scanner_gate/V = holder
+	var/list/status = list()
+	status += "The Green light is [V.pass ? "blinking" : "off"]."
+	status += "The Red light is [V.fail ? "blinking" : "off"]."
+	status += "The Yellow light is [V.triggered ? "blinking" : "off"]."
+	status += "The Purple light is [V.ignore_signals ? "on" : "off"]."
+	return status
```
Comment (Rohesie) [inline code/datums/wires/scan_gate.dm:34]:
```suggestion
	. = list()
	. += "The Green light is [V.pass ? "blinking" : "off"]."
	. += "The Red light is [V.fail ? "blinking" : "off"]."
	. += "The Yellow light is [V.triggered ? "blinking" : "off"]."
	. += "The Purple light is [V.ignore_signals ? "on" : "off"]."
```

Comment (Rohesie) [inline code/datums/wires/scan_gate.dm:34]:
```suggestion
	var/obj/machinery/scanner_gate/scanner = holder
	. = list()
	. += "The Green light is [scanner.pass ? "blinking" : "off"]."
	. += "The Red light is [scanner.fail ? "blinking" : "off"]."
	. += "The Yellow light is [scanner.triggered ? "blinking" : "off"]."
	. += "The Purple light is [scanner.ignore_signals ? "on" : "off"].
```
Try to use better variable names than single upper-case letters.

File: code/game/machinery/scan_gate.dm [hunk 4]
```diff
@@ -39,6 +43,7 @@
 
 /obj/machinery/scanner_gate/Initialize()
 	. = ..()
+	wires = new /datum/wires/scan_gate(src)
```
Comment (Rohesie) [inline code/game/machinery/scan_gate.dm:46]: Be sure to delete and clean the reference in `Destroy()`

File: code/game/machinery/scan_gate.dm [hunk 5]
```diff
@@ -157,12 +168,25 @@
...
 	if(reverse)
 		beep = !beep
 	if(beep)
 		alarm_beep()
-	else
+	if(!ignore_signals)
+		color = wires.get_color_of_wire(WIRE_PROCEED)
+		var/obj/item/assembly/P = wires.get_attached(color)
+		if(istype(P))
+			P.activate()
```
Comment (Rohesie) [inline code/game/machinery/scan_gate.dm:179]: Same here, try to use better variable names than `P` and `F`

**Comment by Rohesie** [comment]
Let us know if you plan to work on this in the future.

**Pull request closed without merging**